### PR TITLE
Remove yum / dnf duplication

### PR DESCRIPTION
**Pull request by sdodson**
@maxamillion FYI

### CODE REVIEW

File: playbooks/common/openshift-cluster/upgrades/v3_0_to_v3_1/upgrade.yml [hunk 1]
```diff
@@ -54,8 +54,8 @@
 - name: Verify upgrade can proceed
   hosts: oo_masters_to_config:oo_nodes_to_config
   tasks:
-  - name: Clean yum cache
-    command: yum clean all
+  - name: Clean yum/dnf cache
```
Comment (brenton) [inline playbooks/common/openshift-cluster/upgrades/v3_0_to_v3_1/upgrade.yml]: can `{{ ansible_pkg_mgr }}` be used in the name?

Comment (abutcher) [inline playbooks/common/openshift-cluster/upgrades/v3_0_to_v3_1/upgrade.yml]: Vars in task names won't be evaluated

Comment (brenton) [inline playbooks/common/openshift-cluster/upgrades/v3_0_to_v3_1/upgrade.yml]: Makes sense.

**Comment by brenton** [comment]
@sdodson, looks like we need a rebase.  Is this something you want to have merged for our async update?

File: playbooks/common/openshift-cluster/upgrades/v3_0_to_v3_1/upgrade.yml (hunk 1 above)
Comment (detiber) [inline playbooks/common/openshift-cluster/upgrades/v3_0_to_v3_1/upgrade.yml]: "Clear package cache"?

**Comment by detiber** [comment]
@brenton async update would be good, the separate tasks for install leave the ability for etcd, master or node corruption in environments with slow storage or network latency.

**Comment by abutcher** [comment]
@dgoodwin FYI

**Comment by sdodson** [comment]
@detiber from the other PR you commented
> We probably want to address the rest of the installs asap. Otherwise the restart handlers for etcd, master and node will be kicked off on the initial config run.
> This can cause errors when the first startup is interrupted with the restart. The restart handler was put in place because of this issue in the first place.

The two cases where I see that we were actually paying attention to the registered result were the iptables and firewalld installation. In all the other cases in the master and node roles we were `register: install_result` but not actually using it so I've removed those.

Am I missing some other way that we're conditionally restarting?

**Comment by detiber** [comment]
Ah, nevermind me, we are testing the service result and not the install result.

**Comment by sdodson** [comment]
> Ah, nevermind me, we are testing the service result and not the install result.

Cool, thanks for checking. I think we're good to merge now.

@brenton looking at the openshift-ansible that's released currently it doesn't have any of the yum/dnf duplication. The iptables/firewalld bits have potential to be a problem so I'd say we merge this for the async release.

**Pull request merged**